x/dex/keeper/grpc_query_limit_order_tranche_user.go

```diff
@@ -46,14 +47,34 @@ func (k Keeper) LimitOrderTrancheUserAll(
 	}, nil
 }
 
+func (k Keeper) CalcWithdrawableShares(ctx sdk.Context, trancheUser types.LimitOrderTrancheUser) (amount math.Int, err error) {
+	tradePairID, tickIndex := trancheUser.TradePairId, trancheUser.TickIndexTakerToMaker
+
+	tranche, _, found := k.FindLimitOrderTranche(
+		ctx,
+		&types.LimitOrderTrancheKey{
+			TradePairId:           tradePairID,
+			TickIndexTakerToMaker: tickIndex,
+			TrancheKey:            trancheUser.TrancheKey,
+		},
+	)
+
+	if !found {
+		return math.ZeroInt(), status.Error(codes.NotFound, "Tranche not found")
+	}
+	withdrawableShares, _ := tranche.Withdraw(&trancheUser)
```

Comment:
How about small refactoring in tranche? we can make new function tranche#Withdrawable (or some kind of this), that will only return stuff and do not modify data. And tranche#Withdraw will use this function and just change `t.ReservesTakerDenom = reservesTokenOutDec.Sub(amountOutTokenOut).TruncateInt()`.
That way there is some boundaries between read and write functions.